include/cocaine/hpack/header_definitions.hpp

```diff
@@ -30,841 +30,846 @@ struct headers {
         }
     };
 
     template<class DefaultValue = default_values_t::empty_string_value_t>
     struct strict_transport_security:
-        public detail::value_mixin<DefaultValue>
+        public DefaultValue
     {
         static
-        constexpr
-        header::data_t
+        const std::string&
         name() {
-            return header::create_data("strict-transport-security");
+            static std::string data("strict-transport-security");
+            return data;
         }
     };
 
     template<class DefaultValue = default_values_t::empty_string_value_t>
     struct transfer_encoding:
-        public detail::value_mixin<DefaultValue>
+        public DefaultValue
     {
         static
-        constexpr
-        header::data_t
+        const std::string&
         name() {
-            return header::create_data("transfer-encoding");
+            static std::string data("transfer-encoding");
+            return data;
         }
     };
 
     template<class DefaultValue = default_values_t::empty_string_value_t>
     struct user_agent:
-        public detail::value_mixin<DefaultValue>
+        public DefaultValue
     {
         static
-        constexpr
-        header::data_t
+        const std::string&
         name() {
-            return header::create_data("user-agent");
+            static std::string data("user-agent");
+            return data;
         }
     };
 
     template<class DefaultValue = default_values_t::empty_string_value_t>
     struct vary:
-        public detail::value_mixin<DefaultValue>
+        public DefaultValue
     {
         static
-        constexpr
-        header::data_t
+        const std::string&
         name() {
-            return header::create_data("vary");
+            static std::string data("vary");
+            return data;
         }
     };
 
     template<class DefaultValue = default_values_t::empty_string_value_t>
     struct via:
-        public detail::value_mixin<DefaultValue>
+        public DefaultValue
     {
         static
-        constexpr
-        header::data_t
+        const std::string&
         name() {
-            return header::create_data("via");
+            static std::string data("via");
+            return data;
         }
     };
 
     template<class DefaultValue = default_values_t::empty_string_value_t>
     struct www_authenticate:
-        public detail::value_mixin<DefaultValue>
+        public DefaultValue
     {
         static
-        constexpr
-        header::data_t
+        const std::string&
         name() {
-            return header::create_data("www-authenticate");
+            static std::string data("www-authenticate");
+            return data;
         }
     };
 
     template<class DefaultValue = default_values_t::zero_uint_value_t>
     struct span_id:
-        public detail::value_mixin<DefaultValue>
+        public DefaultValue
     {
         static
-        constexpr
-        header::data_t
+        const std::string&
```

Comment:
Do we really need to return a const reference to `std::string`. Why not `const char*`?